ChiModules/KEigenvalueSolver/k_eigenvalue_initialize.cc

```diff
@@ -0,0 +1,42 @@
+#include "k_eigenvalue_solver.h"
+#include "ChiMath/SpatialDiscretization/FiniteElement/PiecewiseLinear/pwl.h"
+
+#include <chi_log.h>
+
+extern ChiLog& chi_log;
+
+
+using namespace LinearBoltzmann;
+
+//###################################################################
+void KEigenvalue::Solver::InitializeKSolver()
+{
+  Initialize(); // LinearBoltzmann::Solver initialize
+
+  //======================================== Initialize precursors
+  if (options.use_precursors)
+  {
+    //clear precursor properties
+    num_precursors = 0.0;
+    max_num_precursors_per_material = 0.0;
+
+    //accumulate precursor count and set max
+    for (auto& xs : material_xs)
+    {
+      num_precursors += xs->num_precursors;
+      if (xs->num_precursors > max_num_precursors_per_material)
+        max_num_precursors_per_material = xs->num_precursors;
+    }
+
+
+    if (num_precursors > 0)
```

Comment:
not sure I fully follow. How can we be inside the conditional use_precursors and then later check if n_prec>0. I think I am missing something.

Reply:
@ragusa Yes this was faulty logic on my part. The thinking was to only do anything regarding precursors if the option is turned on and then to only initialize the vector if there are precursors, but that’s not ideal. I think I will modify this to compute the total number of precursors and max per material outside of any conditional, then, if num_precursors == 0, set use_precursors = false, then initialize the vector is use_precursors == true. I am not at my computer today because I am moving so I will get this done in a few days.